amgut/lib/data_access/survey.py

```diff
@@ -353,6 +353,12 @@ def store_survey(self, consent_details, with_fk_inserts,
             of the data to insert
         """
         with TRN:
+            # checks if user has previously been
+            # removed and has still revoked consent
+            sql = """SELECT ag_login_id FROM ag.consent_revoked"""
+            TRN.add(sql)
+            revoked = [result[0] for result in TRN.execute()[0]]
```

Comment:
can this be made a set comprehension since it is used as a lookup? eg

```python
revoked = {result[0] for result in TRN.execute()[0]}
```

Reply:
...actually, this can be done directly against the database by doing something like:

```python
sql = """select exists (select 1 from ag.ag_consent where ag_login_id=%s)"""
TRN.add(sql, [consent_details['login_id']])
exists = TRN.execute()[0]  # I think this is the necessary indexing...
```

Example from the ag test database below:

```sql
ag_test=# select exists (select 1 from ag.ag_consent where ag_login_id='ffffffff-7cf9-2135-e040-8a80115d6401');
 exists
--------
 f
(1 row)

ag_test=# select exists (select 1 from ag.ag_consent where ag_login_id='d8592c74-7cf9-2135-e040-8a80115d6401');
 exists
--------
 t
(1 row)
```

Reply:
Hmm, I'm not sure how this exactly is supposed to work, since I don't quite know how to grab the consent details from the database (I get thrown the error that global name consent_details does not exist).